File: MovieCatalogue_Java_Submission_1/app/src/main/java/id/apwdevs/moviecatalogue/model/OtherAboutFilmModel.java
package id.apwdevs.moviecatalogue.model;

import java.util.ArrayList;

public class OtherAboutFilmModel {
    private String originalLanguage;
    private String longMovieRuntime;
    private String movieRevenue;
    private String movieBudget;
    private String movieGenres;
    private ArrayList<FilmTopBilledCastModel> listTopBilledCast;

    public OtherAboutFilmModel(String sParser, String strList) {
        this(sParser);
        setListTopBilledCast(strList);
    }

    private OtherAboutFilmModel(String sParser) {
        setAll(sParser);
    }

    public OtherAboutFilmModel() {
    }

    public void setAll(String sParser) {
        String[] allComponents = sParser.split("[|]");
        this.originalLanguage = allComponents[0];
        this.longMovieRuntime = allComponents[1];
        this.movieBudget = allComponents[2];
        this.movieRevenue = allComponents[3];
        this.movieGenres = allComponents[4];
    }

    public String getOriginalLanguage() {
        return originalLanguage;
    }

    public void setOriginalLanguage(String originalLanguage) {
        this.originalLanguage = originalLanguage;
    }

    public String getLongMovieRuntime() {
        return longMovieRuntime;
    }

    public void setLongMovieRuntime(String longMovieRuntime) {
        this.longMovieRuntime = longMovieRuntime;
    }

    public String getMovieRevenue() {
        return movieRevenue;
    }

    public void setMovieRevenue(String movieRevenue) {
        this.movieRevenue = movieRevenue;
    }

    public String getMovieBudget() {
        return movieBudget;
    }

    public void setMovieBudget(String movieBudget) {
        this.movieBudget = movieBudget;
    }

    public String getMovieGenres() {
        return movieGenres;
    }

    public void setMovieGenres(String movieGenres) {
        this.movieGenres = movieGenres;
    }

    public ArrayList<FilmTopBilledCastModel> getListTopBilledCast() {
        return listTopBilledCast;
    }

    public void setListTopBilledCast(ArrayList<FilmTopBilledCastModel> listTopBilledCast) {
        this.listTopBilledCast = listTopBilledCast;
    }


    public void setListTopBilledCast(String strList) {

        listTopBilledCast = new ArrayList<>();
        StringBuilder buff = new StringBuilder();
        String buff2 = null;
        for (int x = 0; x < strList.length(); x++) {
            if (strList.charAt(x) == '=') {
                buff2 = buff.toString();
                buff.delete(0, buff.length());
            } else if (strList.charAt(x) == ',') {
                if (buff2 != null) {
                    listTopBilledCast.add(new FilmTopBilledCastModel(buff2, buff.toString()));
                    buff2 = null;
                }
                buff.delete(0, buff.length());
            } else
                buff.append(strList.charAt(x));
        }
        listTopBilledCast.add(new FilmTopBilledCastModel(buff2, buff.toString()));
    }
}

File: MovieCatalogue_Kotlin_Submission_3/app/src/main/java/id/apwdevs/moTvCatalogue/model/OtherAboutFilmModel.kt
package id.apwdevs.moTvCatalogue.model

import java.util.*

data class OtherAboutFilmModel(
    val sParser: String
) {
    init {
        setAll(sParser)
    }

    internal var longMovieRuntime: String? = null
    internal var originalLanguage: String? = null
    internal var movieRevenue: String? = null
    internal var movieBudget: String? = null
    internal var movieGenres: String? = null
    var listTopBilledCast: ArrayList<FilmTopBilledCastModel> = ArrayList()

    private fun setAll(sParser: String) {
        val allComponents = sParser.split("[|]".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        this.originalLanguage = allComponents[0]
        this.longMovieRuntime = allComponents[1]
        this.movieBudget = allComponents[2]
        this.movieRevenue = allComponents[3]
        this.movieGenres = allComponents[4]
    }

    fun setListTopBilledCast(strList: String) {
        if (listTopBilledCast.size > 0)
            listTopBilledCast.clear()
        val buff = StringBuilder()
        var buff2: String? = null
        for (x in 0 until strList.length) {
            when (strList[x]) {
                '=' -> {
                    buff2 = buff.toString()
                    buff.delete(0, buff.length)
                }
                ',' -> {
                    if (buff2 != null) {
                        listTopBilledCast.add(FilmTopBilledCastModel(buff2, buff.toString()))
                        buff2 = null
                    }
                    buff.delete(0, buff.length)
                }
                else -> buff.append(strList[x])
            }
        }
        listTopBilledCast.add(FilmTopBilledCastModel(buff2, buff.toString()))
    }
}